src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,278 @@
 import android.Manifest;
 import android.app.Activity;
 import android.content.Context;
+import android.content.pm.ActivityInfo;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
 
-import com.google.firebase.analytics.FirebaseAnalytics; // FirebaseAnalytics class
+import com.google.firebase.analytics.FirebaseAnalytics;
+import com.google.firebase.analytics.FirebaseAnalytics.Event;
+import com.google.firebase.analytics.FirebaseAnalytics.Param;
 import com.segment.analytics.Analytics;
+import com.segment.analytics.Properties;
 import com.segment.analytics.ValueMap;
 import com.segment.analytics.integrations.IdentifyPayload;
 import com.segment.analytics.integrations.Integration;
 import com.segment.analytics.integrations.Logger;
 import com.segment.analytics.integrations.ScreenPayload;
 import com.segment.analytics.integrations.TrackPayload;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Date;
+import java.util.regex.Pattern;
+
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
 import static com.segment.analytics.internal.Utils.hasPermission;
 import static com.segment.analytics.internal.Utils.isNullOrEmpty;
-import static com.segment.analytics.Analytics.LogLevel.VERBOSE;
+import static com.segment.analytics.internal.Utils.toISO8601Date;
 
 /**
- * Google Analytics for Firebase is a free app measurement solution that provides insight on
- * app usage and user engagement.
+ * Google Analytics for Firebase is a free app measurement solution that provides insight on app
+ * usage and user engagement.
  *
  * @see <a href="https://firebase.google.com/docs/analytics/">Google Analytics for Firebase</a>
  * @see <a href="#">Google Analytics for Firebase Integration</a>
  * @see <a href="#">Google Analytics for Firebase Android SDK</a>
  */
 public class FirebaseIntegration extends Integration<FirebaseAnalytics> {
 
-  public static FirebaseAnalytics mFirebaseAnalytics;
+  public static final Factory FACTORY =
+      new Factory() {
+        @Override
+        public Integration<?> create(ValueMap settings, Analytics analytics) {
+          Logger logger = analytics.logger(FIREBASE_ANALYTICS_KEY);
+          if (!hasPermission(
+              analytics.getApplication(), Manifest.permission.ACCESS_NETWORK_STATE)) {
+            logger.debug("ACCESS_NETWORK_STATE is required for Firebase Analytics.");
+            return null;
+          }
+          if (!hasPermission(analytics.getApplication(), Manifest.permission.INTERNET)) {
```

Comment:
FWIW this is unnecessary. The Segment SDK checks for this permission, so this will always exists.